src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs

```diff
@@ -185,6 +185,13 @@ private async Task<(Index Index, string Collection)> MatchIndex(IndexQueryServer
         {
             var collection = query.Metadata.CollectionName;
 
+            if (query.Metadata.DynamicIndexName != null)
```

Comment:
If we have a new index that supersede the previous index?
For that matter, are we certain that the now shared `Metadata` is safe to share among threads and databases?